### PR TITLE
METAL-1094: Sync with upstream metal3-io/ironic-image

### PR DESCRIPTION
**Pull request by MahnoorAsghar**

### CODE REVIEW

**Comment by MahnoorAsghar** [comment]
I tried to get rid of the Makefile change, with little success ^

**Comment by elfosardo** [comment]
/hold
some more changes are needed if you're removing inspector
check requirements.cachito and prepare-image.sh

**Comment by elfosardo** [comment]
/approve
change looks good
let's see what the CI says
thanks!

**Review by dtantsur — CHANGES_REQUESTED** [review]

File: prepare-image.sh [hunk 1]
```diff
@@ -24,7 +24,6 @@ if  [[ -f /tmp/main-packages-list.ocp ]]; then
     REQS="${REMOTE_SOURCES_DIR}/requirements.cachito"
     IRONIC_UID=1002
     IRONIC_GID=1003
-    INSPECTOR_GID=1004
```
[the remark this answers is not in the document]
Comment (MahnoorAsghar) [reply prepare-image.sh:27]: CBO does not specify the UIDs and GIDs anymore

File: ironic-config/ironic.conf.j2 [hunk 2]
```diff
@@ -136,34 +137,8 @@ external_callback_url = {{ env.IRONIC_EXTERNAL_CALLBACK_URL }}
...
-[inspector]
-# NOTE(dtantsur): we properly configure the "unmanaged" inspection boot (i.e.
-# booting IPA through a separate inspector.ipxe rather than the driver's boot
-# interface), so managed boot is not required.
-require_managed_boot = False
-power_off = {{ false if env.IRONIC_FAST_TRACK == "true" else true }}
-# NOTE(dtantsur): keep inspection arguments synchronized with inspector.ipxe
-# Also keep in mind that only parameters unique for inspection go here.
-# No need to duplicate pxe_append_params/kernel_append_params.
-extra_kernel_params = ipa-inspection-collectors={{ env.IRONIC_IPA_COLLECTORS }} ipa-enable-vlan-interfaces={{ env.IRONIC_ENABLE_VLAN_INTERFACES }} ipa-inspection-dhcp-all-interfaces=1 ipa-collect-lldp=1
```
[the remark this answers is not in the document]
Comment (MahnoorAsghar) [reply ironic-config/ironic.conf.j2:148]: My bad, I thought this was tied to inspector alone.

File: ironic-config/ironic.conf.j2 [hunk 3]
```diff
@@ -136,34 +137,8 @@ external_callback_url = {{ env.IRONIC_EXTERNAL_CALLBACK_URL }}
...
-cafile = {{ env.IRONIC_INSPECTOR_CACERT_FILE }}
-insecure = {{ env.IRONIC_INSPECTOR_INSECURE }}
-{% endif %}
-{% if env.IRONIC_INSPECTOR_CALLBACK_ENDPOINT_OVERRIDE %}
-callback_endpoint_override = {{ env.IRONIC_INSPECTOR_CALLBACK_ENDPOINT_OVERRIDE }}
-{% endif %}
-{% else %}
-hooks = $default_hooks,parse-lldp
-add_ports = all
-keep_ports = present
```
[the remark this answers is not in the document]
Comment (MahnoorAsghar) [reply ironic-config/ironic.conf.j2:162]: Lines 150-155 need to be removed, and 156-164 need to be kept?

[the remark this answers is not in the document]
Comment (MahnoorAsghar) [reply ironic-config/ironic.conf.j2:162]: Edited to remove the if block 

> if env.USE_IRONIC_INSPECTOR == "true"

File: prepare-image.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (MahnoorAsghar) [reply prepare-image.sh:27]: They were only being specified for the inspector container in CBO, which is now removed

**Review by dtantsur — APPROVED** [review]

**Pull request merged**